pkg/server/api/entry/v1/service.go

```diff
@@ -250,3 +280,49 @@ func applyMask(e *types.Entry, mask *types.EntryMask) { //nolint: unused,deadcod
 		e.DnsNames = nil
 	}
 }
+
+func buildListEntriesRequest(req *entry.ListEntriesRequest) (*datastore.ListRegistrationEntriesRequest, error) {
+	listReq := &datastore.ListRegistrationEntriesRequest{}
+
+	if req.PageSize > 0 {
+		listReq.Pagination = &datastore.Pagination{
+			PageSize: req.PageSize,
+			Token:    req.PageToken,
+		}
+	}
+
+	if req.Filter != nil {
+		if req.Filter.ByParentId != nil {
+			var err error
+			listReq.ByParentId, err = api.StringValueFromProto(req.Filter.ByParentId)
+			if err != nil {
+				return nil, fmt.Errorf("malformed ByParentId: %v", err)
+			}
+		}
+
+		if req.Filter.BySpiffeId != nil {
+			var err error
+			listReq.BySpiffeId, err = api.StringValueFromProto(req.Filter.BySpiffeId)
+			if err != nil {
+				return nil, fmt.Errorf("malformed BySpiffeId: %v", err)
+			}
+		}
+
+		if req.Filter.BySelectors != nil {
+			var dsSelectors []*common.Selector
+			for _, reqSelector := range req.Filter.BySelectors.Selectors {
+				s := &common.Selector{
+					Type:  reqSelector.Type,
+					Value: reqSelector.Value,
+				}
+				dsSelectors = append(dsSelectors, s)
+			}
```

Comment:
Maybe this can go into an `api.SelectorsFromProto` that also does the selector type and value validation that we have done elsewhere?

Reply:
Absolutely! Giving these are selectors, I suppose it should go under common and not server.

Reply:
The `api` package is where we've been putting that stuff, since for now, these new types are only part of the server API.

Reply:
i.e. `pkg/server/api`